lnd.go

```diff
@@ -108,7 +136,7 @@ func lndMain() error {
 		net.JoinHostPort("", strconv.Itoa(cfg.PeerPort)),
 	}
 	server, err := newServer(defaultListenAddrs, chanDB, activeChainControl,
-		idPrivKey)
+		idPrivKey, macaroonService)
```

Comment:
Only part of the way through this PR but, at first glance, it seems that the `rpcServer` should be tracking the `macaroonService`, rather than the `server` itself.